test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java

```diff
@@ -176,6 +177,7 @@ public void beforeEach(ExtensionContext context) throws Exception {
         if (isNativeTest(context)) {
             return;
         }
+        pushMockContext();
```

Comment:
This method should probably be called after the `failedBoot` because if we do fail to boot, `pushMockContext` (and `popMockContext` will throw a NPE, thus masking the real problem)